diff --git a/d2l-typography.js b/d2l-typography.js
index b57ea3a..ebd0c20 100644
--- a/d2l-typography.js
+++ b/d2l-typography.js
@@ -1,12 +1,48 @@
 import '@polymer/polymer/polymer-legacy.js';
 import 'd2l-colors/d2l-colors.js';
 import './d2l-typography-shared-styles.js';
+const importUrl = 'https://s.brightspace.com/lib/fonts/0.4.0/assets/';
 const $_documentContainer = document.createElement('template');
 
 $_documentContainer.innerHTML = `<dom-module id="d2l-typography">
 	<template>
 		<style>
-			@import url("https://s.brightspace.com/lib/fonts/0.4.0/fonts.css");
+			@font-face {
+				font-family: 'Lato';
+				font-style: normal;
+				font-weight: 400;
+				src: local('Lato Regular'), local('Lato-Regular'), url(${new URL('Lato-400.woff2', importUrl)}) format('woff2'), url(${new URL('Lato-400.woff', importUrl)}) format('woff'), url(${new URL('Lato-400.ttf', importUrl)}) format('truetype');
+			}
+			@font-face {
+				font-family: 'Lato';
+				font-style: normal;
+				font-weight: 700;
+				src: local('Lato Bold'), local('Lato-Bold'), url(${new URL('Lato-700.woff2', importUrl)}) format('woff2'), url(${new URL('Lato-700.woff', importUrl)}) format('woff'), url(${new URL('Lato-700.ttf', importUrl)}) format('truetype');
+			}
+			@font-face {
+				font-family: 'Open Dyslexic';
+				font-style: normal;
+				font-weight: 400;
+				src: local('Open Dyslexic Regular'), local('OpenDyslexic-Regular'), url(${new URL('OpenDyslexic.woff', importUrl)}) format('woff'), url(${new URL('OpenDyslexic.ttf', importUrl)}) format('truetype');
+			}
+			@font-face {
+				font-family: 'Open Dyslexic';
+				font-style: italic;
+				font-weight: 400;
+				src: local('Open Dyslexic Italic'), local('OpenDyslexic-Italic'), url(${new URL('OpenDyslexic-Italic.woff', importUrl)}) format('woff'), url(${new URL('OpenDyslexic-Italic.ttf', importUrl)}) format('truetype');
+			}
+			@font-face {
+				font-family: 'Open Dyslexic';
+				font-style: normal;
+				font-weight: 700;
+				src: local('Open Dyslexic Bold'), local('OpenDyslexic-Bold'), url(${new URL('OpenDyslexic-700.woff', importUrl)}) format('woff'), url(${new URL('OpenDyslexic-700.ttf', importUrl)}) format('truetype');
+			}
+			@font-face {
+				font-family: 'Open Dyslexic';
+				font-style: italic;
+				font-weight: 700;
+				src: local('Open Dyslexic Bold Italic'), local('OpenDyslexic-BoldItalic'), url(${new URL('OpenDyslexic-700-Italic.woff', importUrl)}) format('woff'), url(${new URL('OpenDyslexic-700-Italic.ttf', importUrl)}) format('truetype');
+			}
 
 			.d2l-typography,
 			.vui-typography {
diff --git a/demo/index.html b/demo/index.html
index 7b8b40e..aae97f7 100644
--- a/demo/index.html
+++ b/demo/index.html
@@ -41,7 +41,7 @@ document.body.appendChild($_documentContainer.content);
 </script>
 		<script type="module">
 const $_documentContainer = document.createElement('template');
-$_documentContainer.innerHTML = `<h1>Typography</h1>`;
+$_documentContainer.innerHTML = '<h1>Typography</h1>';
 document.body.appendChild($_documentContainer.content);
 </script>
 		<script type="module">
diff --git a/test/acceptance/typography.test.js b/test/acceptance/typography.test.js
index 6c74d49..9a4b34f 100644
--- a/test/acceptance/typography.test.js
+++ b/test/acceptance/typography.test.js
@@ -28,8 +28,8 @@ var browsers = {
 	})
 };
 
-var endpoint = 'http://localhost:8000/components/d2l-typography/test/acceptance/typography.html';
-var endpointMixins = 'http://localhost:8000/components/d2l-typography/test/acceptance/typography-mixins.html';
+var endpoint = 'http://localhost:8081/components/d2l-typography/test/acceptance/typography.html';
+var endpointMixins = 'http://localhost:8081/components/d2l-typography/test/acceptance/typography-mixins.html';
 var spec = 'test/acceptance/typography.gspec';
 
 polymerTests(browsers, function(test) {
