diff --git a/cli/iconset-builder.js b/cli/iconset-builder.js
index 3cc8d11..627dc53 100644
--- a/cli/iconset-builder.js
+++ b/cli/iconset-builder.js
@@ -6,10 +6,6 @@ var path = require('path'),
 	fs = require('fs');

 var iconsetObjTemplate = {
-	'link': [
-		{'$': {'rel': 'import', 'href': '../polymer/polymer.html'}},
-		{'$': {'rel': 'import', 'href': '../iron-iconset-svg/iron-iconset-svg.html'}}
-	],
 	'iron-iconset-svg': {
 		'$': {'name': '', 'size': '', 'rtl-mirroring': true, 'use-global-rtl-attribute': true},
 		'svg': {'defs': {'g': []}}
@@ -108,9 +104,14 @@ module.exports = function(iconsetPath) {

 			var deferred = q.defer();

-			xml = xml.substring(7, xml.length - 7);
-
-			fs.writeFile(path.join(__dirname, '../', name + '-icons.html'), xml, function(err) {
+			var content = `import '@polymer/polymer/polymer-legacy.js';
+import '@polymer/iron-iconset-svg/iron-iconset-svg.js';
+const $_documentContainer = document.createElement('template');
+$_documentContainer.setAttribute('style', 'display: none;');
+$_documentContainer.innerHTML = \`${xml}\`;
+document.head.appendChild($_documentContainer.content);
+`;
+			fs.writeFile(path.join(__dirname, '../', name + '-icons.js'), content, function(err) {
 				if (err) {
 					return deferred.reject(err);
 				}
