diff --git a/d2l-input-checkbox-spacer.js b/d2l-input-checkbox-spacer.js
index 162c1e1..cf5d636 100644
--- a/d2l-input-checkbox-spacer.js
+++ b/d2l-input-checkbox-spacer.js
@@ -22,10 +22,6 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-checkbox-spacer">
 				padding-left: 1.7rem;
 				margin-bottom: 0.9rem;
 			}
-			:host-context([dir="rtl"]) {
-				padding-right: 1.7rem;
-				padding-left: 0;
-			}
 			:host(:dir(rtl)) {
 				padding-right: 1.7rem;
 				padding-left: 0;
@@ -33,7 +29,7 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-checkbox-spacer">
 		</style>
 		<slot></slot>
 	</template>
-	
+
 </dom-module>`;
 
 document.head.appendChild($_documentContainer.content);
diff --git a/d2l-input-checkbox.js b/d2l-input-checkbox.js
index 5ff51b9..ba79a0c 100644
--- a/d2l-input-checkbox.js
+++ b/d2l-input-checkbox.js
@@ -14,7 +14,6 @@ import '@polymer/polymer/polymer-legacy.js';
 import 'd2l-colors/d2l-colors.js';
 import 'd2l-typography/d2l-typography-shared-styles.js';
 import { Polymer } from '@polymer/polymer/lib/legacy/polymer-fn.js';
-import { PolymerElement } from '@polymer/polymer/polymer-element.js';
 import { useShadow } from '@polymer/polymer/lib/utils/settings.js';
 const $_documentContainer = document.createElement('template');
 
@@ -43,16 +42,12 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-checkbox">
 				vertical-align: middle;
 				white-space: normal;
 			}
-			:host-context([dir="rtl"]) .d2l-input-checkbox-label {
-				margin-right: 0.5rem;
-				margin-left: 0;
-			}
+
 			:host(:dir(rtl)) .d2l-input-checkbox-label {
 				margin-right: 0.5rem;
 				margin-left: 0;
 			}
-			:host([aria-label]) .d2l-input-checkbox-label,
-			:host-context([dir="rtl"][aria-label]) .d2l-input-checkbox-label {
+			:host([aria-label]) .d2l-input-checkbox-label {
 				margin-left: 0;
 				margin-right: 0;
 			}
@@ -111,7 +106,7 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-checkbox">
 			<span class="d2l-input-checkbox-label"><slot></slot></span>
 		</label>
 	</template>
-	
+
 </dom-module>`;
 
 document.head.appendChild($_documentContainer.content);
@@ -184,15 +179,10 @@ Polymer({
 	},
 	_handleChange: function(e) {
 		this.checked = e.target.checked;
-
-		// in shady DOM the input's "change" event will leak through,
-		// so no need to fire it
-		if (PolymerElement || useShadow) {
-			this.dispatchEvent(new CustomEvent(
-				'change',
-				{bubbles: true, composed: false}
-			));
-		}
+		this.dispatchEvent(new CustomEvent(
+			'change',
+			{bubbles: true, composed: false}
+		));
 	},
 	_handleFocus: function() {
 		// in shady DOM the input's "focus" event does not bubble,
diff --git a/d2l-input-search.js b/d2l-input-search.js
index 2f049e1..e4e8be6 100644
--- a/d2l-input-search.js
+++ b/d2l-input-search.js
@@ -41,12 +41,6 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-search">
 				white-space: nowrap;
 			}
 
-			:host-context([dir="rtl"]) input[type="text"].d2l-input {
-				padding-right: 0.75rem;
-				padding-left: 2.4rem;
-			}
-
-			/* Duplicated because some browsers ignore CSS block for any invalid selector */
 			:host(:dir(rtl)) input[type="text"].d2l-input {
 				padding-right: 0.75rem;
 				padding-left: 2.4rem;
@@ -58,14 +52,6 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-search">
 				padding-right: calc(2.4rem - 1px);
 			}
 
-			:host-context([dir="rtl"]) input[type="text"].d2l-input:hover,
-			:host-context([dir="rtl"]) input[type="text"].d2l-input:focus,
-			:host-context([dir="rtl"]) .d2l-input-search-hover input[type="text"].d2l-input,
-			:host-context([dir="rtl"]) .d2l-input-search-focus input[type="text"].d2l-input {
-				padding-right: calc(0.75rem - 1px);
-				padding-left: calc(2.4rem - 1px);
-			}
-
 			:host(:dir(rtl)) input[type="text"].d2l-input:hover,
 			:host(:dir(rtl)) input[type="text"].d2l-input:focus,
 			:host(:dir(rtl)) .d2l-input-search-hover input[type="text"].d2l-input,
@@ -85,11 +71,6 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-search">
 				right: 0.2rem;
 			}
 
-			:host-context([dir="rtl"]) d2l-button-icon {
-				left: 0.2rem;
-				right: auto;
-			}
-
 			:host(:dir(rtl)) d2l-button-icon {
 				left: 0.2rem;
 				right: auto;
@@ -107,8 +88,6 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-search">
 		</div>
 	</template>
 
-	
-
 </dom-module>`;
 
 document.head.appendChild($_documentContainer.content);
diff --git a/d2l-input-text.js b/d2l-input-text.js
index 449eb88..6fb97d8 100644
--- a/d2l-input-text.js
+++ b/d2l-input-text.js
@@ -16,8 +16,6 @@ import 'd2l-polymer-behaviors/d2l-focusable-behavior.js';
 import './d2l-input-text-behavior.js';
 import './d2l-input-shared-styles.js';
 import { Polymer } from '@polymer/polymer/lib/legacy/polymer-fn.js';
-import { PolymerElement } from '@polymer/polymer/polymer-element.js';
-import { useShadow } from '@polymer/polymer/lib/utils/settings.js';
 const $_documentContainer = document.createElement('template');
 
 $_documentContainer.innerHTML = `<dom-module id="d2l-input-text">
@@ -34,7 +32,7 @@ $_documentContainer.innerHTML = `<dom-module id="d2l-input-text">
 		</style>
 		<input aria-invalid$="[[ariaInvalid]]" aria-label$="[[ariaLabel]]" aria-labelledby$="[[ariaLabelledby]]" autofocus$="[[autofocus]]" class="d2l-input d2l-focusable" disabled$="[[disabled]]" list$="[[list]]" max$="[[max]]" maxlength$="[[maxlength]]" min$="[[min]]" minlength$="[[minlength]]" name$="[[name]]" on-change="_handleChange" on-keypress="_handleKeypress" pattern$="[[pattern]]" placeholder$="[[placeholder]]" readonly$="[[readonly]]" required$="[[required]]" size$="[[size]]" step$="[[step]]" tabindex$="[[tabIndex]]" type$="[[type]]" value="{{value::input}}">
 	</template>
-	
+
 </dom-module>`;
 
 document.head.appendChild($_documentContainer.content);
@@ -113,13 +111,11 @@ Polymer({
 		}
 	},
 	_handleChange: function() {
-		if (PolymerElement || useShadow) {
-			// Change events don't automatically propagate across shadow DOM boundaries
-			this.dispatchEvent(new CustomEvent(
-				'change',
-				{bubbles: true, composed: false}
-			));
-		}
+		// Change events don't automatically propagate across shadow DOM boundaries
+		this.dispatchEvent(new CustomEvent(
+			'change',
+			{bubbles: true, composed: false}
+		));
 	},
 	_handleKeypress: function(e) {
 		if (this.preventSubmit && e.keyCode === 13) {
diff --git a/d2l-input-textarea.js b/d2l-input-textarea.js
index 8d4e8b8..d6b3916 100644
--- a/d2l-input-textarea.js
+++ b/d2l-input-textarea.js
@@ -27,8 +27,6 @@ import './d2l-input-text-behavior.js';
 import './d2l-input-shared-styles.js';
 import { Polymer } from '@polymer/polymer/lib/legacy/polymer-fn.js';
 import { dom } from '@polymer/polymer/lib/legacy/polymer.dom.js';
-import { PolymerElement } from '@polymer/polymer/polymer-element.js';
-import { useShadow } from '@polymer/polymer/lib/utils/settings.js';
 const $_documentContainer = document.createElement('template');
 
 $_documentContainer.innerHTML = `<dom-module id="d2l-input-textarea">
@@ -306,13 +304,11 @@ Polymer({
 	},
 
 	_handleChange: function() {
-		if (PolymerElement || useShadow) {
-			// Change events don't automatically propagate across shadow DOM boundaries
-			this.dispatchEvent(new CustomEvent(
-				'change',
-				{bubbles: true, composed: true}
-			));
-		}
+		// Change events don't automatically propagate across shadow DOM boundaries
+		this.dispatchEvent(new CustomEvent(
+			'change',
+			{bubbles: true, composed: true}
+		));
 	}
 
 });
diff --git a/demo/d2l-input-textarea.html b/demo/d2l-input-textarea.html
index 231185d..2ed3cb3 100644
--- a/demo/d2l-input-textarea.html
+++ b/demo/d2l-input-textarea.html
@@ -20,7 +20,7 @@
 		-->
 	<script type="module">
 const $_documentContainer = document.createElement('template');
-$_documentContainer.innerHTML = `<style is="custom-style" include="d2l-typography"></style>`;
+$_documentContainer.innerHTML = '<style is="custom-style" include="d2l-typography"></style>';
 document.head.appendChild($_documentContainer.content);
 </script>
 		<script type="module">
diff --git a/test/d2l-input-checkbox.html b/test/d2l-input-checkbox.html
index 8886e73..b36ad05 100644
--- a/test/d2l-input-checkbox.html
+++ b/test/d2l-input-checkbox.html
@@ -33,8 +33,6 @@
 		<script type="module">
 import '@polymer/iron-test-helpers/mock-interactions.js';
 import '../d2l-input-checkbox.js';
-import { PolymerElement } from '@polymer/polymer/polymer-element.js';
-import { useShadow } from '@polymer/polymer/lib/utils/settings.js';
 describe('d2l-input-checkbox', function() {
 
 	var elem;
@@ -123,11 +121,7 @@ describe('d2l-input-checkbox', function() {
 			it('should fire "change" event when input element is clicked', function(done) {
 				var input = elem.$$('input');
 				elem.addEventListener('change', function(e) {
-					if (PolymerElement || useShadow) {
-						expect(e.target).to.equal(elem);
-					} else {
-						expect(e.target).to.equal(input);
-					}
+					expect(e.target).to.equal(elem);
 					done();
 				});
 				MockInteractions.tap(input);
