diff --git a/d2l-search-widget-behavior.js b/d2l-search-widget-behavior.js
index 9a41821..dcee864 100644
--- a/d2l-search-widget-behavior.js
+++ b/d2l-search-widget-behavior.js
@@ -1,5 +1,5 @@
 import '@polymer/polymer/polymer-legacy.js';
-import 'siren-parser/siren-parser.js';
+import SirenParse from 'siren-parser';
 window.D2L = window.D2L || {};
 window.D2L.PolymerBehaviors = window.D2L.PolymerBehaviors || {};
 
@@ -121,7 +121,7 @@ D2L.PolymerBehaviors.SearchWidgetBehavior = {
 		}
 
 		// Siren parser can only parse entities, so make a fake one with the given action
-		var entity = window.D2L.Hypermedia.Siren.Parse({
+		var entity = SirenParse({
 			actions: [ searchAction ]
 		});
 		var parsedSearchAction = entity.actions[0];
@@ -170,7 +170,7 @@ D2L.PolymerBehaviors.SearchWidgetBehavior = {
 				}
 				return Promise.reject(response.status + ' ' + response.statusText);
 			})
-			.then(window.D2L.Hypermedia.Siren.Parse)
+			.then(SirenParse)
 			.then(this._onSearchResponse.bind(this));
 	}
 
diff --git a/package.json b/package.json
index 0760a1f..5fe84ba 100644
--- a/package.json
+++ b/package.json
@@ -3,7 +3,7 @@
   "name": "d2l-search-widget",
   "scripts": {
     "lint": "npm run lint:wc && npm run lint:js",
-    "lint:js": "eslint . --ext .js,.html test/**/*.js test/**/*.html demo/**/*.js demo/**/*.html",
+    "lint:js": "eslint . --ext .js,.html test/**/*.html demo/**/*.html",
     "lint:wc": "polymer lint",
     "test:polymer:local": "polymer test --skip-plugin sauce",
     "test:polymer:sauce": "polymer test --skip-plugin local",
